### ISSUE TITLE
bug: `production` service resource requires idle_scaling attributes, but can't disable it

### ISSUE THREAD

**Issue by parkedwards** [ClickHouse/terraform-provider-clickhouse#24]
Provider version: 0.0.3

note: i'm only showing the relevant attributes for the resource, as we use locals to configure each of our services

```terraform
# fails configuration validation, as it requires idle_timeout_minutes to be set
# even if idle_scaling is disabled
{
      tier                 = "production"
      idle_scaling         = false
}
```

Error message (upon plan):
> idle_scaling, min_total_memory_gb, max_total_memory_gb, and idle_timeout_minutes must be defined if the service tier is production

```terraform
# to bypass the above, I set idle_timeout_minutes to the default value
{
      tier                 = "production"
      idle_scaling         = false
      idle_timeout_minutes = 5
}
```

Error message (upon apply):
> When applying changes to clickhouse_service.events, provider "provider[\"registry.terraform.io/clickhouse/clickhouse\"]" produced an unexpected new value: .idle_scaling: was cty.False, but now cty.True.

In the second case, the service was created, but it looks like setting `idle_timeout_minutes = 5` to bypass resource config validation would actually enable idle_scaling, so there's an inconsistent API value when rehydrating state

![image](https://github.com/ClickHouse/terraform-provider-clickhouse/assets/19556538/85a0acf9-891e-447b-afab-b7c0cc66de46)

**Comment by parkedwards** [comment]
this is similar to what is described in this issue: https://github.com/ClickHouse/terraform-provider-clickhouse/issues/5#issuecomment-1644098248

**Comment by Kinzeng** [comment]
this should be addressed in #33

**Comment by guykoh** [comment]
@Kinzeng is this fixed now? Can we close this issue?
